### DISCUSSION TITLE
GeoMap centre option and dynamic GeoJSON filename

### DISCUSSION THREAD

**Discussion by erlfos** [grafana/grafana#46287] [Dashboards, Panels & Visualizations]
Hi
The Geomap plugin looks useful, but it has one issue that prevent me from moving on from the Worldmap plugin and that is the option to centre the map on the "Last GeoHash" (the option in the Worldmap panel) 
![image](https://user-images.githubusercontent.com/45815831/157028904-675ff4fe-87a9-4068-a4f4-2550081a6c01.png)

or some way to use variables in the longitude and latitude fields in the "Initial view". With this option in the Worldmap panel I am able to centre the map on a new location if I change the location using a variable for example.
![image](https://user-images.githubusercontent.com/45815831/157028499-dc6aa520-5fc2-43ee-9348-230e708ddbc8.png)

I would also like to add a feature request for the Geomap plugin, and that is the ability to change the GeoJSON file dynamically using variables so that I can change the GeoJSON file when I select a new location using variables, like this:
![image](https://user-images.githubusercontent.com/45815831/157028739-63cb96d1-cc46-4cbb-b91f-60d3bdc5c57b.png)

**Comment by mrzone64** [comment]
I also request this feature for the Geomap plugin.

**Comment by geoffgpe** [comment]
The ability to initially centre the map using variables would be most welcome. An essential feature in many ways.

**Comment by sesoko838** [comment]
Centering the map based on data layers seems to be implemented in Grafana 8.5.3 and is working for me when using a SQL or Influx query
![image](https://user-images.githubusercontent.com/106267858/171266427-2c950a0e-d99b-4d61-bc08-db78fbbc632f.png)

Definitely would like a variable for the GeoJSON

**Reply by erlfos** [reply]
Centering the map with "Fit data layers" only works for the initial loading of the dashboard. If I change a variable so that latitude and longitude change it will not center on the new location, although the new location is visible on the map. This is Grafana v8.5.4

**Comment by tedraymond** [comment]
Most definitely. I have 100s maybe 1000s of geojson files of data that would be really nice to use with variables. Users would love this addition.  Can we make this happen

**Comment by luizluca** [comment]
Variable support in any geomap source (gazetteer or geojson) would be really nice.

BTW, please add variables support everywhere, on any field!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83000. Feel free to continue the discussion around this there. Thank you!
